main.cpp

```diff
@@ -3922,21 +3922,22 @@ static picoboot::connection get_single_rp2350_bootsel_device_connection(device_m
 
 struct progress_bar {
     explicit progress_bar(string prefix, int width = 30) : prefix(std::move(prefix)), width(width) {
+        // Align all bars with the longest possible prefix string
+        auto longest_mem = std::max_element(
+            std::begin(memory_names), std::end(memory_names),
+            [] (const auto & p1, const auto & p2) {
+                return p1.second.length() < p2.second.length();
+            }
+        );
+        longest_mem_str = longest_mem->second;
         progress(0);
     }
 
     void progress(int _percent) {
         if (_percent != percent) {
             percent = _percent;
             unsigned int len = (width * percent) / 100;
-            // Align all bars with the longest possible prefix string
-            auto longest_mem = std::max_element(
-                std::begin(memory_names), std::end(memory_names),
-                [] (const auto & p1, const auto & p2) {
-                    return p1.second.length() < p2.second.length();
-                }
-            );
-            string extra_space(string("Loading into " + longest_mem->second + ": ").length() - prefix.length(), ' ');
+            string extra_space(string("Loading into " + longest_mem_str + ": ").length() - prefix.length(), ' ');
```

Comment:
The `string("Loading into " + longest_mem_str + ": ").length()` could also be done just once? :stuck_out_tongue_winking_eye:

Reply:
Fixed by adding the extra space before setting the prefix

Reply:
Excellent, cheers.